### PR TITLE
ISPN-13277 Log exceptions for IRAC

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-13277

### CODE REVIEW

**Comment by pruivo** [comment]
@oraNod if you have time, can you take a look at the log messages and "upgrading" guide? thanks!

File: core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java:34]:
```suggestion
    * @param site Specifies the name of the backup location for this cache. The name must match a site defined in the 
    *             global configuration.
```

File: core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java:75]:
```suggestion
    * Sets the strategy used for backing up data: sync or async. Defaults to {@link
    * org.infinispan.configuration.cache.BackupConfiguration.BackupStrategy#ASYNC}.
```

File: core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java:95]:
```suggestion
    * Configures the policy for handling failed requests for synchronous backups. Defaults to 
    * {@link org.infinispan.configuration.cache.BackupFailurePolicy#WARN}
```

File: core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java:103]:
```suggestion
    * Controls if replication happens in a two phase commit (2PC) for synchronous backups. The default value is "false", which means replication happens in a one phase commit (1PC).
```

File: core/src/main/java/org/infinispan/util/logging/Log.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/util/logging/Log.java:772]:
```suggestion
   @Message(value = "Encountered issues while backing up data for cache %s to site %s", id = 202)
```

File: core/src/main/java/org/infinispan/util/logging/Log.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/util/logging/Log.java:1198]:
```suggestion
   @Message(value = "You must specify a 'failure-policy-class' to use a custom backup failure policy for backup '%s'!", id = 338)
```

File: core/src/main/java/org/infinispan/util/logging/Log.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/util/logging/Log.java:2230]:
```suggestion
   @Message(value = "The backup '%s' configuration 'failure-policy=%s' is not valid with an ASYNC backup strategy.", id = 655)
```

File: core/src/main/java/org/infinispan/util/logging/Log.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/util/logging/Log.java:2233]:
```suggestion
   @Message(value = "The backup '%s' configuration 'failure-policy-class' is not compatible with 'failure-policy=%s'. Use 'failure-policy=\"CUSTOM\"'", id = 656)
```

File: core/src/main/java/org/infinispan/xsite/irac/IracResponseCollector.java
Comment (oraNod) [inline core/src/main/java/org/infinispan/xsite/irac/IracResponseCollector.java:75]:
```suggestion
               log.tracef(throwable, "Encountered issues while backing up data for cache %s to site %s", cacheName, backup.getSiteName());
```

File: core/src/test/java/org/infinispan/xsite/XSiteCacheConfigurationTest.java
Comment (oraNod) [inline core/src/test/java/org/infinispan/xsite/XSiteCacheConfigurationTest.java:89]:
```suggestion
   @Test(expectedExceptions = CacheConfigurationException.class, expectedExceptionsMessageRegExp = "ISPN\\d+: You must specify a 'failure-policy-class' to use a custom backup failure policy for backup 'LON-1'.")
```

File: documentation/src/main/asciidoc/topics/upgrading.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/upgrading.adoc:8]:
```suggestion
By default, the asynchronous backup strategy logs exceptions that occur on remote sites.
To disable exceptions from backup locations and log exceptions for the local site only, set `failure-policy="IGNORE"` in your configuration as follows:
```

**Review by oraNod — COMMENTED** [review]
Some suggestions here @pruivo Let me know if you want another round or disagree with any of these changes. Cheers.

File: documentation/src/main/asciidoc/topics/upgrading.adoc
Comment (pruivo) [reply documentation/src/main/asciidoc/topics/upgrading.adoc:8]: @oraNod actually, `and log exceptions for the local site only` do not make much sense since we don't cross-site replicate to itself.

File: core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java
Comment (pruivo) [reply core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java:95]: I'll remove the word `synchronous` since not it support both sync and async

File: documentation/src/main/asciidoc/topics/upgrading.adoc
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/upgrading.adoc:8]: @pruivo That makes sense. I made some edits to the suggestion.

**Review by oraNod — APPROVED** [review]
LGTM @pruivo Do you need more approvals before merging?

**Comment by pruivo** [comment]
@oraNod go for it :)

**Comment by oraNod** [comment]
thanks @pruivo

**Pull request merged**
